src/components/cards/Item.tsx

```diff
@@ -49,80 +53,10 @@ const Item: React.FC<PropTypes> = ({ data, category, history }) => {
     }, [])
 
     /**
-     * Adds the item to the shopping list
+     * Handles adding item to shopping list
      */
-    const addItemToShopList = () => {
-        setShopList((current: any) => {
-            const currentItem: ItemType = {
-                ...data,
-                quantity: 1,
-                categoryName: category,
-            }
```

Comment:
Is there a reason for you to delete that part from the hook? Seems to be the cause of the issue mentioned ;)